src/main/java/LogicHandlers/CommandHandlers/EventHandler.java

```diff
@@ -0,0 +1,64 @@
+package LogicHandlers.CommandHandlers;
+
+import Exceptions.DukeInvalidDateTimeException;
+import Exceptions.DukeInvalidFormatException;
+import Models.Event;
+import Models.TaskArray;
+
+import static LogicHandlers.Parsers.DateTimeParser.parseDateTimeFromString;
+import static Ui.BasicOutputPrinter.printBasicOutput;
+import static Ui.ErrorOutputPrinter.printErrorOutput;
+
+/**
+ * EventHandler handles all 'event' commands.
+ */
+public class EventHandler implements Command {
+
+    private TaskArray tasks;
+
+    /**
+     * Constructor for EventHandler.
+     *
+     * @param tasks The TaskArray we are working with
+     */
+    public EventHandler(TaskArray tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Parses the content of the input.
+     *
+     * @param commandContent The content of the input.
+     */
+    @Override
+    public void parseCommandContent(String commandContent) {
+        if (commandContent.equals("")) {
+            printErrorOutput("You cannot add an empty 'Event' task!");
+        } else {
+            try {
+                String[] taskArr = commandContent.split("/", 3);
+
+                if (taskArr.length != 3) {
+                    String errorStr = "\nFormat for Event task incorrect!\n" +
+                            "Expected 3 parts (Task name, start & end time) in input, got " + taskArr.length;
+                    throw new DukeInvalidFormatException(errorStr);
+                }
+
+                tasks.addTask(new Event(taskArr[0].strip(), false,
+                        parseDateTimeFromString(taskArr[1].strip()), parseDateTimeFromString(taskArr[2].strip())));
```

Comment:
woah this code is quite trippy to read. Probably can split into different expressions to ease readability:

```java
String eventDescription = taskArr[0].strip();
LocalDateTime fromDate = parseDateTimeFromString(taskArr[1].strip());
LocalDateTime toDate = parseDateTimeFromString(taskArr[2].strip());

tasks.addTask(new Event(eventDescription, false, fromDate, toDate));
```